admin-frontend/app_singleapp/lib/widgets/setup/page1_widget.dart

```diff
@@ -238,31 +248,38 @@ class _SetupPage1ThirdPartyProviders extends StatelessWidget {
       required this.bloc,
       required this.selectedExternalProviderFunc})
       : super(key: key);
-
   @override
   Widget build(BuildContext context) {
     final external = bloc.has3rdParty;
-
-    final children = <Widget>[];
-
     if (external) {
-      bloc.externalProviders.forEach((provider) {
-        children.add(InkWell(
-          mouseCursor: SystemMouseCursors.click,
-          onTap: () {
-            selectedExternalProviderFunc(provider);
-          },
-          child: Image.asset(bloc.externalProviderAssets[provider]!),
-        ));
-        children.add(Padding(
-          padding: const EdgeInsets.fromLTRB(0, 16, 0, 10),
-          child: Text('or enter your details to register',
-              style: Theme.of(context).textTheme.caption),
-        ));
-      });
+      return Column(
+        children: [
+          for (dynamic provider in bloc.externalProviders)
+            Container(
+              height: 48,
+              child: Padding(
+                padding: const EdgeInsets.only(top: 12.0),
+                child: SignInButton(
+                    provider == 'oauth2-google'
+                        ? Buttons.GoogleDark
+                        : provider == 'oauth2-github'
+                        ? Buttons.GitHub
+                        : Buttons.Microsoft,
+                    text: provider == 'oauth2-google'
+                        ? ('Register with Google')
+                        : provider == 'oauth2-github'
+                        ? ('Register with GitHub')
+                        : ('Register with Microsoft'),
+                    onPressed: () {
+                  selectedExternalProviderFunc(provider);
+                }),
+              ),
+            ),
+        ],
+      );
+    } else {
+      return SizedBox();
```

Comment:
if you put this at the top and just say if (!external) return SizedBox.shrink(); Then it de-indents the whole rest of the section and makes it much easier to follow. I've changed this in a number of places to decomplexify the logic.